build/rust/download_rust_toolchain_aux.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import glob
+from urllib.error import HTTPError
+import os
+from pathlib import Path
+import shutil
+import tempfile
+
+import brave_chromium_utils
+import build_rust_toolchain_aux
+import deps
+import deps_config
+
+with brave_chromium_utils.sys_path(build_rust_toolchain_aux.TOOLS_RUST):
+    import build_rust
+    RUST_TOOLCHAIN = build_rust.RUST_TOOLCHAIN_OUT_DIR
+
+BRAVE_PREFIX = '.brave-'
+PACKAGE = os.path.join(
+    RUST_TOOLCHAIN, f'{BRAVE_PREFIX}{build_rust_toolchain_aux.package_name()}')
+PATTERN = os.path.join(RUST_TOOLCHAIN, f'{BRAVE_PREFIX}*')
+
+
+def is_download_needed():
+    return os.getenv('SKIP_DOWNLOAD_RUST_TOOLCHAIN_AUX',
+                     '0') != '1' and not Path(PACKAGE).exists()
+
+
+def maybe_remove_old_package():
+    if entries := glob.glob(PATTERN):
+        [entry] = entries
+        print(f'Removing {entry}')
+        os.unlink(entry)
+
+
+def install_new_package(src_dir):
+    print(f'Installing {PACKAGE}')
+
+    rust_lld_path = os.path.join(RUST_TOOLCHAIN, 'bin',
+                                 build_rust_toolchain_aux.RUST_LLD)
+    wasm32_unknown_unknown_path = os.path.join(
+        RUST_TOOLCHAIN, 'lib', 'rustlib',
+        build_rust_toolchain_aux.WASM32_UNKNOWN_UNKNOWN)
+
+    # delete existing artifacts
+    Path(rust_lld_path).unlink(missing_ok=True)
+    shutil.rmtree(wasm32_unknown_unknown_path, ignore_errors=True)
+
+    # move new artifacts to their final places
+    shutil.move(os.path.join(src_dir, build_rust_toolchain_aux.RUST_LLD),
+                rust_lld_path)
+    shutil.move(
+        os.path.join(src_dir, build_rust_toolchain_aux.WASM32_UNKNOWN_UNKNOWN),
+        wasm32_unknown_unknown_path)
+
+
+def main():
+    if not is_download_needed():
+        print(f'Skipping {Path(__file__).stem}')
+        return
+
+    maybe_remove_old_package()
+
+    with tempfile.TemporaryDirectory(dir=RUST_TOOLCHAIN) as temp_dir:
+        try:
+            deps.DownloadAndUnpack(
+                f'{deps_config.DEPS_PACKAGES_URL}/rust-toolchain-aux/{build_rust_toolchain_aux.package_name()}',
+                temp_dir,
+                output_file=PACKAGE)
```

Comment:
sure, you can keep the whole archive and make some changes to `DownloadAndUnpack`, but consider storing a simple text file like `${RUST_TOOLCHAIN}/.brave_toolchain_aux_package` with the downloaded/unpacked package and read/write it from you script to detect if you need the update.